File: AppCheck/AppCheck/app/src/main/java/jp/gr/java_conf/bunooboi/appcheck/MainActivity.java
package jp.gr.java_conf.bunooboi.appcheck;

import android.app.ActivityManager;
import android.content.Context;
import android.content.Intent;
import android.content.pm.ApplicationInfo;
import android.content.pm.PackageManager;
import android.graphics.Bitmap;
import android.graphics.BitmapFactory;
import android.graphics.Point;
import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.view.Display;
import android.view.WindowManager;
import android.widget.TextView;

import java.util.List;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.main);

        stopService(new Intent(getApplicationContext(), MainService.class));
        startService(new Intent(getApplicationContext(), MainService.class));

        TextView textview = (TextView) findViewById(R.id.textview);
        textview.setText("禁止アプリを起動");
        textview.setTextSize(40 * getScaleSize(getApplicationContext()));
    }

    @Override
    public void onResume() {
        super.onResume();
        ActivityManager activityManager = ((ActivityManager) getSystemService(ACTIVITY_SERVICE));
        PackageManager pm = getPackageManager();
        List<ApplicationInfo> appList = pm.getInstalledApplications(0);

        for (int i = 0; i < appList.size(); i++) {
            String packageName = appList.get(i).packageName;
            if (packageName.equals("com.google.android.youtube")) {
                activityManager.killBackgroundProcesses(packageName);
            }
        }
    }

    public static float getScaleSize(Context context) {// 文字サイズ調整
        Bitmap bm = BitmapFactory.decodeResource(context.getResources(), R.drawable.saizu);
        WindowManager wm = (WindowManager) context.getSystemService(Context.WINDOW_SERVICE);
        Display disp = wm.getDefaultDisplay();
        Point size = new Point();
        disp.getSize(size);
        float width = 0;
        float scale = 0;
        width = size.x;
        scale = (float) width / (float) bm.getWidth();
        return scale;
    }
}

File: MemorySportsSmart/app/src/main/java/jp/gr/java_conf/bunooboi/memorysportssmart/MainActivity.kt
package jp.gr.java_conf.bunooboi.memorysportssmart

import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        Main.init()

        val board1: Board = findViewById(R.id.board1)
        val board2: Board = findViewById(R.id.board2)

        board1.textView.text = "Memory"
        board2.textView.text = "Training"

        board1.imageButton.setImageResource(R.mipmap.memory)
        board2.imageButton.setImageResource(R.mipmap.training)

        board1.imageButton.setOnClickListener {
            Main.methodType = Main.TYPE_MEMORY
            startActivity(Intent(this, SelectCompetitionActivity::class.java))
        }

        board2.imageButton.setOnClickListener {
            Main.methodType = Main.TYPE_TRAINING
            startActivity(Intent(this, SelectCompetitionActivity::class.java))
        }
    }
}
